### PR TITLE
TM-5289-Position-Classification

### PR DESCRIPTION
**Pull request by cagbakwu**
[Ticket](https://metaphase.atlassian.net/browse/TM-5289)

Before : Re-rendering all card components tabs:
https://github.com/MetaPhase-Consulting/State-TalentMAP/assets/135657303/db752252-00e8-4dd4-b8b5-99e3f8262884

After : Rendering specific card components tab:
https://github.com/MetaPhase-Consulting/State-TalentMAP/assets/135657303/5039b243-b538-4141-a9c7-d9efd1597ede

Data saves properly on the backend, It was more-so a visual bug.

### CODE REVIEW

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
 
   const results = useSelector(state => state.positionClassifications);
+  const currentCard = useSelector(state => state.positionClassificationsNumber);
   const isLoading = useSelector(state => state.positionClassificationsIsLoading);
   const pc = results?.positionClassifications ?? [];
   const cs = results?.classificationSelections ?? [];
 
   useEffect(() => {
     if (positionNumber) {
+      dispatch(positionClassificationNumberCheck(positionNumber));
```
Comment (cagbakwu) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:19]: Saved current position number when selecting specific card

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
-import { positionClassifications, positionClassificationsEdit } from '../../../actions/positionClassifications';
+import { positionClassificationNumberCheck, positionClassifications, positionClassificationsEdit } from '../../../actions/positionClassifications';
 
 const PositionClassification = (props) => {
   const { positionNumber, bureau, posSeqNum } = props;
 
   const dispatch = useDispatch();
 
   const results = useSelector(state => state.positionClassifications);
+  const currentCard = useSelector(state => state.positionClassificationsNumber);
```
Comment (cagbakwu) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:12]: new state that I added to redux.

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 3]
```diff
@@ -67,7 +69,7 @@ const PositionClassification = (props) => {
     }
   };
 
-  return (isLoading ?
+  return (isLoading && positionNumber === currentCard ?
```
Comment (cagbakwu) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:72]: Loading state will only occur for the card with the specific position number.

**Comment by elizabeth-jimenez** [comment]
I see changes made to account for the loading, but there were also data changes happening on the cards that also needs to be addressed; has it been addressed? You may need to modify mock data in order to see different classifications per card

**Comment by cagbakwu** [comment]
> I see changes made to account for the loading, but there were also data changes happening on the cards that also needs to be addressed; has it been addressed? You may need to modify mock data in order to see different classifications per card

@elizabeth-jimenez yes, I had a teams call with Meagan and we saw that the data is saving properly on the backend even after refreshing, the correct checkboxes are being checked. The classifications update every time a user clicks the classification tabs of any card.

**Comment by cagbakwu** [comment]
@elizabeth-jimenez ill verify again this morning to make sure im correct 🤣 .

**Pull request merged**